src/firebase_token.rs

```diff
@@ -0,0 +1,72 @@
+//! Extracting bearer tokens from request headers
+//! and with the assumption that they're Firebase JWTs, checking
+//! their validity.
+
+use crate::{errors::{Error, InvalidAuthHeader, InvalidJwt}, DecodedToken, FirebaseAuth};
+use rocket::{request::{Request, FromRequest, self}, http::Status};
+
+/// The bearer token included in request headers
+///
+/// Valid requests with the `Authorization` header included should look
+/// like the following:
+/// ```ignore
+/// Authorization: Bearer <some_bearer_token>
+/// ```
+#[derive(Debug)]
+pub struct FirebaseToken {
+    // The JWT token with all its claims
+    pub token: DecodedToken,
+}
+
+/// Try to convert an Authorization headers to a valid bearer token
+#[rocket::async_trait]
+impl<'r> FromRequest<'r> for FirebaseToken {
+    type Error = Error;
+
+    async fn from_request(req: &'r Request<'_>) -> request::Outcome<Self, Self::Error> {
+        // Is header present?
+        match req.headers().get_one("Authorization") {
+            Some(header_raw) => {
+                let (header_name, header_content) = header_raw.split_at(7);
```

Comment:
It might make more sense to change `src/plugins/rocket.rs` instead of creating a new file since we already have a `FromRequest` trait implementation.